### ISSUE TITLE
App unavailable on non-German Play Store

### ISSUE THREAD

**Issue by AHDGraham** [corona-warn-app/cwa-app-android#478]
## Your Question

* Source File: N/A
* Line(s): N/A
* Question: Congrats on your launch. Problem: Google Play store shows my device as incompatible with the app. (One+ 6). I suspect you are limited by Google checking the owner's Google account region. My account is UK but I am permanently living in Germany (resident, citizen, permanent job). If you cannot rectify this, then your app coverage will be limited. It will not be usable by those residents that probably travel the most! These are surely the very people the app needs to track the most. I think a solution to this is therefore needed asap. (As always, apologies if this issue/question is posted elsewhere - I did look before posting  ;)     )

**Comment by boxwood** [comment]
Similar issue for me. Swiss play store account. But as a German resident and weekly commuter I should use both apps (Swiss and German) as long there is no code roaming solution in place.

**Comment by SebastianWolf-SAP** [comment]
Thank you for your message. We are clarifying this with the Robert Koch Institute. I’m sorry for the inconvenience caused by only publishing it in the German Play Store. We keep you updated once we have any news.

Mit freundlichen Grüßen/Best regards, 
SW
Corona Warn-App Open Source Team

**Comment by mat-l** [comment]
I have a similar problem. Please find a screenshot of the error message below. Google deleted some services from my Huawei phone. I installed them manually again, but your app is not working.

Will you publish the app to Huawei App Gallery with their services as well?

![Screenshot_20200616_080709](https://user-images.githubusercontent.com/5571947/84737809-f3ed4080-afa8-11ea-9c41-5f7519b84ffa.jpg)

**Comment by SvenLieber** [comment]
I have a similar issue in Belgium. When visiting the play store from the Browser I see the error message `This app is incompatible with all of your devices`.
When visiting the play store from my PlayStore app I see the, luckily more concrete, error message `This item is not available in your country`.

**Comment by abhimishr** [comment]
Google play recommends switching countries. 
Play Store - Hamburger Menu on left right - Account - Preferences - Country and profiles

Switching to Germany won't show `This item is not available in your country` message anymore. This is problematic on its own because[ Google doesn't allow you changing it more than once a year](https://support.google.com/googleplay/answer/7431675?hl=en) and you'll loose your Google Play credits as well. In EU where we see quite a few people moving around between borders this cannot be a agreed upon solution.  [From an earlier comment by Sebastian, they are aware of this issue.](https://github.com/corona-warn-app/cwa-app-android/issues/478#issuecomment-644549577)

**Comment by corneliusroemer** [comment]
Is it not possible to sideload the APK? From outside the play store? This has worked for me in the past with similar country restrictions.

**Comment by boxwood** [comment]
> Is it not possible to sideload the APK? From outside the play store? This has worked for me in the past with similar country restrictions.
> 
> This may work: https://apkpure.com/de/corona-warn-app/de.rki.coronawarnapp

If you have a trustworthy APK yes. but this should be just a temporary workaround. you dont want to sideload from shady apk mirrors. i see no reason why the app shouldn't be published globally?
i guess the architecture can handle this.

**Comment by arianvp** [comment]
Note that Google actually limits you to only moving one time per year. Which is of course a horrible constraint if you're an EU resident.  I currently live in Germany but I cant change my Google accounts country because Google has decided that moving multiple countries per year is not allowed.

Lifting this restriction seems wise to me. Thanks for all the work!

**Comment by corneliusroemer** [comment]
> > Is it not possible to sideload the APK? From outside the play store? This has worked for me in the past with similar country restrictions.
> > 
> > This may work: https://apkpure.com/de/corona-warn-app/de.rki.coronawarnapp
> 
> If you have a trustworthy APK yes. but this should be just a temporary workaround. you dont want to sideload from shady apk mirrors. i see no reason why the app shouldn't be published globally?
> i guess the architecture can handle this.

I think SAP has said they don't want to support sideloading. So I guess you have to trust it. It should still be signed correctly. Otherwise the Exposure API won't work if I'm not wrong. Otherwise there would be no testing issue for non-whitelisted Devs.

**Comment by corneliusroemer** [comment]
> Thank you for your message. We are clarifying this with the Robert Koch Institute. I’m sorry for the inconvenience caused by only publishing it in the German Play Store. We keep you updated once we have any news.
> 
> Mit freundlichen Grüßen/Best regards,
> SW
> Corona Warn-App Open Source Team

All these issues could have been raised earlier had you allowed a beta test of interested members of the public.

Good to hear it seems only a bureaucratic mistake - and not one imposed by Google. So please fix this asap.

**Comment by Flo082002** [comment]
Maybe it is possible to upload the apk in the releases here on github

**Comment by lennarto** [comment]
On your iPhone you are able to use more than one Apple ID per Device - Therefore I have a US-Apple ID as well as a german AppleID to download specific apps.

**Comment by boxwood** [comment]
> On your iPhone you are able to use more than one Apple ID per Device - Therefore I have a US-Apple ID as well as a german Apple ID to download specific apps.

very useful hint for android users ;-) but maybe senor google will consider such a flexibility in future releases. like IOS supports multi user scenarios (not)

**Comment by L-E** [comment]
My whole family has the same problem as we moved a few month ago from the UK to Germany and our phones are still stuck on the UK Google Play store. Despite talking to Google support I haven't been able to manually change the country. And Google only allows this a maximum of once a year anyway. 

RKI should change the country distribution for the app in the Play store to all countries to avoid that someone travelling to Germany or living in Germany isn't able to download the app.

**Comment by cooldaniel1991** [comment]
Same problem with the Italian play store. Don't forget Germany has 10 million foreigners.

**Comment by caugner** [comment]
Workaround:

1. Install F-Droid. 
2. Install Aurora Store via F-Droid.
3. Open Aurora and use "Anonymous" login.
4. Change location to "Germany / Berlin".
5. Restart app.
6. Search for "corona warn app rki".

**Comment by RexJarvis** [comment]
Same problem, UK Play Store - Please just publish the official APK.

**Comment by kurthuwig** [comment]
Same here in France. As the play store uses the country of the credit card and you cannot change it easily - and not at all if you have a family account.

It does not make any sense to restrict the app to any country. The virus is not restricted by borders, so why is the app?

**Comment by tegk** [comment]
Please fix this immediately!

**Comment by gcampionpae** [comment]
Same issue here :(

**Comment by schmitzhenninglmu** [comment]
As an Apple/iOS user I have an analogue issue. The app is unavailable in non-German store.

**Comment by corneliusroemer** [comment]
How long will this take to fix? Should be only one click. This issue has been known for more than 6 hours. To estimate conservatively, 1% of potential users have this problem and half of them won't come back. Half of them will have tried installing by now. So we'll have already lost around 100k users to this avoidable bug so far. Get going!

**Comment by flennic** [comment]
Same issue here. Living in Sweden and have to travel to Germany. Considering the current situation here that should be fixed asap.

Is there any plan to publish on F-Droid and/or is that possible at all?

**Comment by arianvp** [comment]
@flennic i don't think F-droid is possible as the app relies on Google Play Services to provide the  exposure notification API.

**Comment by flennic** [comment]
> @flennic i don't think F-droid is possible as the app relies on Google Play Services to provide the  exposure notification API.

Ah okay, I wasn't sure if it just relies on services generally available on the phone or if that is related the app store, as I didn't spend time investigating this. Thanks for clarifying!

**Comment by corneliusroemer** [comment]
@flennic I think you may have a quarantine problem anyways. Unless the Bundesland you go to doesn't enforce quarantine or you have a valid exemption.

**Comment by flennic** [comment]
> @flennic I think you may have a quarantine problem anyways. Unless the Bundesland you go to doesn't enforce quarantine or you have a valid exemption.

Yes, I will have to do that, but that is not the point of the issue :) Just wanted to make sure and emphasize that it would be good if the app is available at least within the European stores.

**Comment by treysis** [comment]
Why do developers even come up with the idea to impose a region lock?

**Comment by kurthuwig** [comment]
Kam gerade rein:

_"Das Unternehmen SAP prüft aktuell, ob die App europaweit/weltweit eingestellt werden kann. Dies wird momentan rechtlich durch SAP geprüft."_

Da hat jemand wirklich nicht das Internet verstanden. Was muss da rechtlich geprüft werden, ob man es für die Europäischen Staaten freischalten darf???

**Comment by corneliusroemer** [comment]
> Kam gerade rein:
> 
> _"Das Unternehmen SAP prüft aktuell, ob die App europaweit/weltweit eingestellt werden kann. Dies wird momentan rechtlich durch SAP geprüft."_
> 
> Da hat jemand wirklich nicht das Internet verstanden. Was muss da rechtlich geprüft werden, ob man es für die Europäischen Staaten freischalten darf???

 🤦 
Und wieder 50k unwiederbringlich verloren.

Mann mann mann, das ist lächerlich. Die Geo-Restriction besteht doch nicht aus juristischen Gründen.

Auf @SAP, dafür könnt ihr doch die 20mio endlich gut einsetzen bei einer Spitzenkanzlei, die euch das in 30min bestätigt.

**Comment by codingcatgirl** [comment]
Das Problem an dieser Stelle ist einfach mal wieder, dass Entwickler das Geolocking falsch verstanden haben. Beim Geolocking geht es nicht um den Aufenthaltsort, sondern um die Rechnungsadresse im Google Play Store, die sich nicht beliebig oft ändern lässt. 

Selbes Problem bei Apps der DHL, diversen Taxi Apps, etc… man schießt sich damit selbst ins bein.

Liebe RKI und SAP: Die App ist bereits im Ausland verfügbar, wenn die Rechnungsadresse stimmt. Was auch immer ihr genau damit erreichen wollt, Geolocking im Play Store macht nicht das, was ihr denkt.

**Comment by corneliusroemer** [comment]
@codingcatgirl Das hätte jeder denen erzählen können der schon zig Mal im Ausland war.  Dass das RKI das nicht weiß ok. Aber SAP? Jeder der mal eine Android App gemacht hat sollte die Einstellung kennen. 

Stand in keinem Dokument, dass die App georestricted sein würde. Also ist das doch auch nicht vorgeschrieben. Und warum muss man das dann jetzt prüfen wenn es davor noch nicht geprüft war? Fragen über Fragen. Wo sind die Journalisten, die mal ein bisschen Druck aufbauen?

**Comment by EccoApp** [comment]
As far as I know Google and Apple want to ensure that there is only one Corona App per country. This could be a reason for the implementation of Geo Blocking.

**Comment by corneliusroemer** [comment]
@EccoApp Correct, one app per country. But that's enforced through explicit whitelisting of apps, not georestricting play accounts. 
Italy's app can be installed here in Germany: https://play.google.com/store/apps/details?id=it.ministerodellasalute.immuni&hl=en

**Comment by vijay399** [comment]
Download from Here if Getting Incompatible on Play store : [https://vliveapk.com/corona-warn-app/](https://vliveapk.com/corona-warn-app/)

**Comment by jbauerrfid** [comment]
The idea of the European Union is to have unrestricted transfer of goods, services, people and finance transaction. In the case of apps this falls into the service category.
Why is this limited for the Corona Warn app? Does SAP still live in the last century? They should unblock it at least for all EU countries quickly.

**Comment by MartinKuhne** [comment]
I have the same issue. We live in the US and Germany. I am unable to change the play store location at this time.

**Comment by corneliusroemer** [comment]
3h since my last comment.

@SebastianWolf-SAP please give us an update. This is probably the number one issue at the moment. First you said you were asking the RKI. What did they say? 
Then someone claimed SAP is checking legality. Any news on that?

Italy, Switzerland, Latvia aren't restricting. Why are we? If this was thought about before, why wasn't it discussed in the scoping document, architecture etc. If it wasn't discussed before, then it wasn't a legal issue since lawyers must have greenlighted it. So what's stopping you?

Has no one read this article yet?
https://support.google.com/googleplay/android-developer/answer/7550024?hl=en
 
I hate to bring this up again but you're getting paid 20mio of tax payer money for 30 Devs (according to GitHub). That's 200k per month. Even if you're ten times as many, it's a ridiculous pay. Surely someone can figure this out.

And I'm sure there's someone available 24/7 at @Google somewhere around the world, willing to assist you with this little Play Store issue.

**Comment by hartenfels** [comment]
Ran into this issue. My Play Store location defaulted to USA for no discernible reason. The Play Store page told me that the app wasn't compatible with my device. Spent a long time trying to figure out what about my phone was inacceptable.

Turns out the reason I couldn't install the app is because I first had to give Google my credit card number. That changed my Play Store region to Germany, which in turn suddenly made my device compatible.

The Corona-Warn-App silently requiring me to give my credit card to Google before I'm allowed to use it is outright moon logic. Telling me that my device has anything to with it would have caused a less persistent person to just give up and not install the app, misleading them into thinking they need a new phone.

**Comment by corneliusroemer** [comment]
@hartenfels What you're talking about is a known workaround - it's not possible for everyone and there are good reasons why the geo-lock should be lifted.

@AHDGraham It was brought to my attention that you labelled this issue a "question". I suggest you relabel it a "bug" to escalate it to its proper status, given the attention it has received.

**Comment by AHDGraham** [comment]
Hi SebastianWolf,
Good to see the Corona Warn-App Open Source Team is dealing with this and I see you have flagged it as a bug. I'd be happy to beta-test any update as a non-google.de account holder.

**Comment by SebastianWolf-SAP** [comment]
Hey @AHDGraham , 

we need to get some legal stuff sorted out, but the required people are definitely aware of the issue and working on it. You know, German laws are sometimes special in comparison to others. As said above, we take care. In the heat of the moment today, we probably simply forgot this 'bug' assignment, no bad intentions...

@corneliusroemer, I need to _repeatedly_ remind you of our [code of conduct](https://github.com/corona-warn-app/cwa-documentation/blob/master/CODE_OF_CONDUCT.md). Your behavior - not only in this thread - showed many elements which we consider to be derogatory or simply inappropriate in a professional setting. We would really like to avoid to speak out warnings or introduce more serious actions. Instead, let's treat each other with courtesy to get the issues fixed.

Thank you very much!

Mit freundlichen Grüßen/Best regards, 
SW
Corona Warn-App Open Source Team

**Comment by fhajji** [comment]
Same problem here. My Galaxy S5 LineageOS + Gapps is connected to the German Playstore, and was able to download the app without issues (from a US IP address).

The Galaxy S4 LineageOS + Gapps of my dad (also from a US IP address) is connected to the US Playstore, and there, Corona Warn App is not even listed (is the S4 supported at all?). The RKI link provided by @corneliusroemer said on the S4 that this app is not available in your country.

Physically, we're both with our phones in Germany right now, but for some reasons we use US IP addresses, so network-wise, we're in the US.

I know there are workarounds, including VPNs and open wifi spots here to temporarily get a German IP address for the sake of downloading the app, but frankly, I won't bother. At most, I'd install on the S4 an official APK from this GitHub project if it were provided, as a last resort. Going the F-Droid route seems too risky, IMHO (e.g. what's the SHA-256 of the official Playstore APK?).

Please fix this availability issue.

**Comment by kurthuwig** [comment]
> > Kam gerade rein:
> > _"Das Unternehmen SAP prüft aktuell, ob die App europaweit/weltweit eingestellt werden kann. Dies wird momentan rechtlich durch SAP geprüft."_
> @kurthuwig wo kam das rein? Presse?

Nein, das war eine Antwort von **CoronaWarnApp <CoronaWarnApp@rki.de>** auf meine Anfrage, warum es nicht im Ausland geht. Die Hotline wusste auch nichts von dem Problem, hat es aber _"an die Vorgesetzte eskaliert"_.

**Comment by MartinKuhne** [comment]
> I know there are workarounds, including VPNs and open wifi spots here to temporarily get a German IP address for the sake of downloading the app, but frankly, I won't bother. At most, I'd install on the S4 an official APK from this GitHub project if it were provided, as a last resort. Going the F-Droid route seems too risky, IMHO (e.g. what's the SHA-256 of the official Playstore APK?).

This is not actually a workaround. I am on T-Mo DSL in Germany and my play store has reverted to the US, and it's not showing or allowing to install.

**Comment by fhajji** [comment]
Ah, good to know. Thanks @MartinKuhne. So we'll have to wait until the app becomes available on the US Play Store.

**Comment by tegk** [comment]
@SebastianWolf-SAP Perhaps you would like to share some more information with us why it is not possible to clarify such a simple issue within one working day. Personally, I simply cannot understand it and would be happy to receive a detailed answer. My family and I are not able to use the app and the chance is high that it will never be used if there is no immediate action.

**Comment by xxyyz** [comment]
Just one comment: one can change the country without changing any credit card data. So if the once-per-a-year limit is not a problem for you, then you can install the app. The steps:

Use any web browser. You can do the first 5 steps below from your laptop.

1) Go to https://myaccount.google.com/ -- you'll have to log in with your google username and password: they are your gmail address and its password

2) On the left hand side you see a menu. Select the last item: "Payments and subscriptions".

3) Select the first item in the middle of the new page: "Payment methods".

4) Select the last, rightmost item from the new menu on the top: "Settings".

5) Here you see your "Payment profile". The second item is the "Country/region", which you can simply change to Germany.

Now you are ready with your Google / Play store payment country change. The last step is to select it on your phone. So you have to do the following steps on your mobile.

6) Start the Play Store app.

7) Tap the menu (top left) and select Account (Konto).

8) Here you should see "Country and profiles" with two countries: your current one (which is not Germany) and Germany. Select Germany.

After this step the installation problem "Dieser Artikel ist in deinem Land nicht verfügbar" should not come anymore and you should be able to install the Corona-Warn-App from https://play.google.com/store/apps/details?id=de.rki.coronawarnapp

**Comment by corneliusroemer** [comment]
Why are you just addressing @AHDGraham and ignoring other points raised by the community? Or people who opened similar issues but whose issues got closed as duplicates? It sounds like you are listening only very selectively.

> @corneliusroemer, I need to _repeatedly_ remind you of our [code of conduct](https://github.com/corona-warn-app/cwa-documentation/blob/master/CODE_OF_CONDUCT.md). Your behavior - not only in this thread - showed many elements which we consider to be derogatory or simply inappropriate in a professional setting. We would really like to avoid to speak out warnings or introduce more serious actions. Instead, let's treat each other with courtesy to get the issues fixed.

@SebastianWolf-SAP Since you make public accusations I should have the right to a defense. But your allegations need to be concrete, not just "in various places". Can you please detail them so that I can understand what you consider derogatory? Isn't it normal for discussions to get heated occasionally?

Are you now using any of the clauses in the code of conduct against me or not? It sounds like you're trying to intimidate me rather than use an actual warning. I prefer clarity. Which articles do you invoke against me if any and why in case you do? I'm sure the community is interested and happy to judge.

This is the first time you mention the code of conduct. Why do you then say repeatedly?

I never intended to hurt anyone. If I did, I apologise. I admit mistakes but I expect the same of others.

On a more general note:

You see, you're getting paid, we're not. All the community cares about is the success of the app. The same cannot be said of SAP. You are making a profit, making money, want to have a good reputation. All understandable. But it means there's is a conflict of interest. If SAP was doing this pro bono, I would say many things very differently. But you are not.

All my criticism and sometimes harsh words were only ever directed at SAP, not the community. You are not a community leader. You are a company employee defending company decisions. If we were a community, there would be some degree of democracy, but there's none of it. It seems like you're stealing the goodwill of open source while maintaining hard us and them boundaries.  

You are silencing inconvenient voices. Shouldn't you have a thicker skin because you get paid? Isn't that part of why you are paid and we are not?

If I feel that you are ignoring the second most important problem voiced by hundreds of people today because it "got lost" this makes me upset. If you say you will update us but then don't for more than half a day this is upsetting. If you close issues as duplicates of issues you've closed as "will do later" this is upsetting.

All of us in the community are spending time here for the public good. We deserve to be treated with respect. And yes, respect means not closing issues just because you don't want to deal with the work. It means keeping your promises and keeping us updated regularly. Having enough staff monitor issues so that you can explain decisions and bring order into chaos.

You can't just cherry pick the community's good sides. If you want us to provide quality issues, feedback, suggestions and trust you have to do your part and respect our needs.

If you want to be celebrated and cheered only and prohibit criticism, then do it explicitly. I'm sure the media and the taxpaying public will find it interesting.

Anyways, this is getting too long. I'd be curious to hear what others are thinking. If it's just me I'll shut up. But reading between the lines I don't think @tegk, @kurthuwig and others are very happy either.

**Comment by piratenkatzerrr** [comment]
*shaking my head* 
When the idea and requirements for the German corona warning app were being discussed in the public, to broaden its appeal the app was being touted as not to require a Google account. 
We are discussing workarounds around a geolock due to account country settings, when people should be able to download and install the app without even having a Google account and hence no account settings.

**Comment by salakis** [comment]
Unbelievable stupid decision in the first place to geoblocks this application. The particular risk group of commuters is being entirely neglected. I hope that this gets fixed ASAP and doesn't end up stuck in some bureaucratic limbo for weeks.

**Comment by coezbek** [comment]
Possible workaround until this gets resolved: create second google account with German address on the phone.

**Comment by MartinKuhne** [comment]
I would like to thank the developers for taking up the feedback and taking steps to address the issue.

**Comment by flennic** [comment]
I would also like @SebastianWolf-SAP for talking on this issue and all your work done so far. If you have further information about this issue, it would be nice if you let us know.

@corneliusroemer While I think it is okay to point out issues with the app, I don't see any reason for your saucy behaviour, even if you don't agree with some decisions. In my opinion, this is not the right place for discussing these kinds of problems. You also have to consider in which short amount of time the application has been released and that yesterday way release day which means there is a heck of things to handle. There will be issues. But they can be addressed appropriately. I also don't think that immediately ranting on Twitter or other GitHub issues is advantageous. Also, degenerating @SebastianWolf-SAP to "you are a company employee defending company decisions" is not kind considering he is a human being like all of us and I assume you don't know him personally to justify this kind of behaviour. Notice, I am not saying your statements are wrong per default, but how you address them is a bit of an issue in my opinion.

I will not further participate in this "personal" discussion and will focus on the technical side. But I felt this had to be said, considering you specifically asked for feedback.

**Comment by gerritholl** [comment]
I have just installed the app and I don't think I have the German Play Store.  Under Play Store -> Preferences -> Country and profiles, no country is listed at all, just "Switch to the Germany play store" is listed, implying I don't have the Germany play store at all.  I have no payment information associated to my Google Account, perhaps this issue is only a problem for users who do?

**Comment by fhajji** [comment]
A little update: I've sent a query to the RKI regarding this issue, and in their reply, RKI confirmed what @SebastianWolf-SAP said: SAP is looking into the legal issues to make Corona WarnApp globally available.

In other words: this is obviously a legal and/or bureaucratic snafu, there is nothing we can do as developers at this point except waiting or trying some workarounds.

@corneliusroemer, while I understand your frustration, please keep it civil and professional. If the issue doesn't resolve by itself in the near future, we can all take it to the German Government / Bundestag, which is ultimately responsible for the legal framework which causes this unfortunate geoblocking issue. @SebastianWolf-SAP is doing his best to help us here.

**Comment by corneliusroemer** [comment]
Thanks @flennic for your extensive comment.

I'll do my best to stay objective. And I will address the maintainers in the future and use @SebastianWolf-SAP only as a generic mention  meaning "SAP maintainers" to make it non-personal.

This is the wrong forum for general criticism of the organisation. Suffice it to say, there was a month to anticipate and solve this problem, those who should have thought about it didn't. Now it's up for the lawyers to solve it and the media to criticise.

Now I'll get back to finding and fixing technical issues.

**Comment by sschuldenzucker** [comment]
@SebastianWolf-SAP agreeing with everything @corneliusroemer said. I don't want anyone threatened with vague references to "some of his behavior" and Github's code is conduct. A bit AfD style tbh. :/ (I came here because I got the same problem, of course)

**Comment by dalbre** [comment]
Hey everybody,
special thanks @corneliusroemer for being perseverant in his demand, which is pure straightforward logic. I cannot see any "saucy" whatsoever and if the point of this forum is to simply and plainly shut down suggestions that should have been part of a common sense requirements assessment in the first place, I believe, we are all wasting our time.

So, we do not have a European corona app for the reasons we all know. 

The national apps do not talk to each other. 

Some national apps can be downloaded in other countries.

And one national cannot even do that. 

What this app should protect against has obviously come from abroad at some point, so it does escape any logic to not include users of countries visiting Germany *and* who are willing to go through the effort of researching, installing and using the app. It's a no-brainer to foresee this functionality and every second spent on discussing the need and holding reasonable contributors to account over their perceived language is an obvious waste of time and effort urgently required for making these various European apps talk to each other before the holiday season.

**Comment by treysis** [comment]
I don't buy this "required by Germany's particularly weird laws". Our laws aren't that different, and after all a lot is covered by EU legislation.

This was just a mishap by the maintainers (which aren't necessarily the developers) who didn't think sufficiently about the impact of their actions. You can observe this negligence with other app vendors (e.g. Swiss PostFinance doesn't think there could be account holders residing outside Switzerland that would like to manage their account on their phone - while there is no restriction to manage it from a PC).

I am 100% convinced someone just thought 'well, this app is for Germany, right? So let's tick this checkbox!'.

**Comment by dalbre** [comment]
@treysis that's exactly it!

**Comment by fhajji** [comment]
@treysis, I don't think so. If RKI and SAP, independently of each others, are mentioning legal issues, then they are probably right.

This doesn't look like an oversight by the dev-team. If it were so, they would have fixed the issue by now in an expedient, non-bureaucratic way, by un-ticking the geo-lock checkbox. It looks like it runs deeper than this, i.e. into the muddy realm of byzantine laws and regulations.

We don't know exactly what the show-stopper is, but if you're interested, lawmakers and politicians do have contact addresses. We'll lobby them ask them in due time what's going on, if this issue persists. But before doing this, let's give RKI and SAP some time to sort that mess out by themselves.

**Comment by treysis** [comment]
@fhajji Independently? I don't think so. You can come up with a lot of excuses if you need a scapegoat ;) If not, what are those legal issues? So far, this has been a very vague statement. Why am I also reacting a bit harsh on this (please remember, this is not personal)? Because I really dislike region locks and they don't do what developers think, and in this particular case it's even more counterproductive.
Ask them in due time? When? In two months? Time? This is a non-issue that could be solved within a day. It's not like all people suddenly have to quit whatever they're doing. Assign one person to this "bug" and make her/him solve it.

**Comment by corneliusroemer** [comment]
@fhajji it's an oversight that this didn't get flagged way in the past to be resolved by now. No one thought of it.

It does seem to be a bureaucratic issue now, but that could have been done in the past.

Since it is a bureaucratic issue, pressure does work! It's urgent so we should lobby NOW.

The maintainers have a conflict of interest to cover up their oversight. So they may not escalate as quickly as would be in the interest of the public.

I'm with @dalbre, @treysis, @sschuldenzucker 

We can't trust the legality explanation without further explanation to support, this could just be a convenient excuse/smoke grenade for why it's not the maintainers fault.

Disclaimer: I'm not blaming any single person. This is not personal. It is also not unprofessional to represent the public interest.

**Comment by kbobrowski** [comment]
This is an exact place in Google Play console to set up country availability, I guess someone who was uploading APK just selected Germany here, instead all countries. I hardly believe there is a legal requirement to allow only people with German Play Store to participate in contact tracing or use any other app developed by SAP / RKI. Play Store country has **nothing to do** with citizenship / residence / tax residence etc, especially in Europe where it is very easy to have multiple bank accounts in couple of countries / multiple residences and just assign some Google products like Play Store to countries which are most convenient for some purpose.

![rki](https://user-images.githubusercontent.com/28542783/84903699-3cebe480-b0d0-11ea-982c-a733ee5a476b.jpg)

**Comment by fhajji** [comment]
@treysis, I'm obviously not happy about this stupid geo-lock. My dad is in his eighties, and has diabetes type II, so he's even more at risk then myself. That his phone can't install this app while my phone can, is mindbogglingly stupid, considering that we live together and therefore count as one, as far as the risk of infection goes. This is definitively not a scenario that was intended by the politicians that are promoting this app.

Speaking from experience, I think, a week or two would be a good measure to wait. That's what it usually takes in such big organizations to reach a decision. After all, they need feedback from lawyers etc, and this does take time. It's not like our workflow, where we quickly fix bugs bugs.

**Comment by kbobrowski** [comment]
But let's not try to judge institutions / specific people who were responsible for this screw-up, I'm sure it's very stressful time for everyone involved and mistakes like this unfortunately happen, we are all only people

**Comment by corneliusroemer** [comment]
@kbobrowski Interesting view, so you are suggesting that the legality is after all a lie? Who knows. I'm not sure
It's not primarily important who is to blamed of course! But this needs to be resolved asap. And  it seems there is a lack of accountability towards the community. Otherwise, why no update all day today?

@fhajji A week or two? I disagree, should have been fixed within 24hr, at most. If not, provide update on progress.

Dear maintainers, please provide an update on this issue. Or at least an update when there will be an update. @SebastianWolf-SAP @tkowark @jakobmoellersap

**Comment by kbobrowski** [comment]
@corneliusroemer I'm suggesting that if there is a legal requirement which limits apps developed by SAP / RKI only to "German Play Store" then legislative body which implemented this requirement is extremely incompetent, since having "German Play Store" does not imply German citizenship / residence or tax residence, and not having "German Play Store" does not mean that someone is not German citizen / resident or tax resident.

Since I believe that we have competent legislative bodies - it's hard to believe in any such legal requirement on the apps.

**Comment by treysis** [comment]
@fhajji You could sideload the app, but that would exclude it from future updates via PlayStore.

**Comment by arianvp** [comment]
Also if such a legal requirement is indeed there which leads to processing of certain personal data (your billing address), it should have been part of the DPIA.

However the DPIA assessment says the following about the app store install procedure: the only  data that is processed is "limitations on age" and "login credentials for the respective app store".  Billing address restrictions is not mentioned as a required processing of data in the DPIA in order to install the app.
![image](https://user-images.githubusercontent.com/628387/84904398-8d9f1580-b0af-11ea-9b47-c9bdbc4a1e0f.png)

https://www.coronawarn.app/assets/documents/cwa-datenschutz-folgenabschaetzung.pdf

If it were a requirement to have Google process and check your billing information (which is what they basically enabled now) this should've been something that must have been explicitly noted in the DPIA; just like the other data processing Google performs  (age, login data, IP addresses during download from the play store).


Either way. I'm sure people are hard at work to get this fixed. I'll try to stay positive and constructive here. However  I agree that this being a "legal requirement" sounds very fuzzy to me.

**Comment by fhajji** [comment]
@corneliusroemer : `The maintainers have a conflict of interest to cover up their oversight.`

What I fail so see is this: what would be the interest of the maintainers to geo-lock the app, and even worse, to keep it geo-locked? The app makers have already been paid, and the app is free to use. What would be the point in geo-locking it?

If we speculated that the Government made a contract with the makers which specifically geo-locked the app (why would they do that?), then the maintainers would have absolutely no leeway whatsoever to unlock this app anyway, even if they wanted to. No amount of complaining here would change anything, and we'll have to take it to the politicians and to the press. But we don't know that yet, so let's give em some time to resolve the issue among themselves. Maybe they are currently negotiating with the Government to update their contract to lift the geo-locks... and we all know how long THAT would take.

So my point is: this is a multi-organizations project, and such projects are by nature slow to react. We have to live with this unfortunate fact of life.

**Comment by courtiol** [comment]
FYI: I am in Germany and I did manage to trick the Play store to display the app by adding it to the wishlist on the web, and then going to the wishlist inside the Play store on my phone. It does show the App but displays: "This item is not available in your country" despite location functionalities being activated for the Play store... How can one check which Play store on uses and how to switch between Play store. Does anyone know?

**Comment by piratenkatzerrr** [comment]
The communication around this product rollout is... a bit of a clusterf*uck? 
The CWA website is hard to find, it appears nobody thought it neither necessary to do some Google SEO, nor to have links set from the RKI's (https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/WarnApp/Warn_App.html) or Bundesregierung's website (https://www.bundesregierung.de/breg-de/themen/corona-warn-app) or vice versa. 
On the other hand, the CWA website is the only site displaying the hotline phone numbers (but only if you scroll aaaaalll the way to the bottom...).
The Bundesregierung's site has FAQs and explainers in sign language and easy language. Which the RKI's and CWA's site don’t. 
The FAQs on the various website explain how the app works, but all lack a troubleshooting section and also do not address questions that have been raised here since early last morning regarding e.g. what data Goggle gathers through the required location activation (in plain layman's terms/language) or why the app is only available through the German Google Play Store (and what to do about that) or how to proceed contact tracing app wise, if you are frequently in several EU countries.
 
If you do go to the Play Store, try to install the app, run into problems and click on the link for developer’s website in hope of getting help, the first thing you see on the CWA website is a referrer to GitHub. 
And with all due respect to GitHub, it’s not the right place for consumers who just want their consumer product/app to work and are looking for a quick fix to their problem or answer to their question, preferably in German, without having to have prior knowledge or spending a lot of time acquiring said knowledge. 

The moment this app went live, an accompanying consumer needs focused website should have gone live with it. One that integrated accessibility needs (sign language, easy language) and troubleshooting. 
Just because it's OpenSource, doesn't mean you have to turn a product launch into an accidental nationwide beta test and let the friendly (unpaid) folks on GitHub handle customer support.

**Comment by tkowark** [comment]
Sorry, but we have to lock this conversation for now. @SebastianWolf-SAP already told you that we're working on a resolution and sometimes these things just take time.

Suggesting cover ups or using profane language doesn't help the matter and are in violation of our [code of conduct](https://github.com/corona-warn-app/cwa-app-android/blob/master/CODE_OF_CONDUCT.md).

Again, we're doing everything we can to resolve this issue and will let you know once we have a solution. Until then, we kindly ask for your patience.

**Comment by tkowark** [comment]
Finally we have an official statement from the Robert Koch-Institute which is also reflected in our FAQ: https://www.coronawarn.app/en/faq/#international and https://www.coronawarn.app/de/faq/#international

> After a detailed examination of the situation it was determined that for legal reasons any publication of the Corona-Warn-App in App Stores outside of Germany is currently only possible after a case-by-case assessment. Legal consultations have shown that in the case of publication in international App Stores, the law of the respective country must be considered and applied to the Corona-Warn-App. This applies in particular to data protection, any necessary claims for information by local authorities and other contractual and consumer protection regulations.
> 
> The Robert Koch-Institute as publisher of the app will trigger this check for the respective countries and release the app after successful legal examination. In a first step, the RKI will release the app for some European countries. These are: Netherlands, Belgium, Luxembourg, France, Austria, Czech Republic, Poland, Denmark, Romania and Bulgaria. It may still take a few hours to a few days until the app is available for the individual countries. We kindly ask you for your patience.
> 
> More countries are currently being checked and will be released in the App Stores step by step. The Robert Koch-Institute asks for your understanding that it cannot provide any information on specific countries and the status of their release during the ongoing review process.

We will move this issue to corona-warn-app/cwa-backlog#9 and add comments for each country or set of countries which are added over time.

Mit freundlichen Grüßen/Best regards,
TK
Corona Warn-App Open Source Team
